File: 2. Codigos/2. Python CPU/poisson/analyticalPoisson.py
import numpy as np

def analytical_poisson(X):
    """
    Returns the analytical solution u(x, y) for validation or Dirichlet BCs.
    
    Parameters
    ----------
    X : ndarray of shape (nPoints, 2)
        Array of (x, y) coordinates where the solution is evaluated.
    
    Returns
    -------
    u : ndarray
        The solution u(x, y) evaluated at each point.
    """
    lambda_ = 10
    f = 6

    x = X[:, 0]
    y = X[:, 1]

    # Simple solution
    u = x**2

    # Complex option (uncomment if needed)
    # u = 4*y**2 - 4*lambda_**2 * y * np.exp(-lambda_ * y) * np.cos(f * np.pi * x) + lambda_ * np.exp(-2 * lambda_ * y)

    return u

File: 2. Codigos/1. Matlab CPU/poisson/analyticalPoisson.m
function u = analyticalPoisson(X)

% Parameters
lambda = 10;
f = 6;

% Points
x = X(:,1);
y = X(:,2);

% Solution
u=x.^2;
%u = 4*y.^2 - 4*lambda^2*y.*exp(-lambda*y).*cos(f*pi*x) + lambda*exp(-2*lambda*y);


